crates/server-primitives/src/admin.rs

```diff
@@ -286,116 +368,78 @@ impl JoinContextRequest {
 }
 
 #[derive(Copy, Clone, Debug, Deserialize, Serialize)]
-#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
 pub struct JoinContextResponseData {
     pub context_id: ContextId,
     pub member_public_key: PublicKey,
 }
 
-impl JoinContextResponseData {
-    #[must_use]
-    pub const fn new(context_id: ContextId, member_public_key: PublicKey) -> Self {
-        Self {
-            context_id,
-            member_public_key,
-        }
-    }
-}
-
 #[derive(Copy, Clone, Debug, Deserialize, Serialize)]
-#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
 pub struct JoinContextResponse {
     pub data: Option<JoinContextResponseData>,
 }
 
 impl JoinContextResponse {
-    #[must_use]
-    pub const fn new(data: Option<JoinContextResponseData>) -> Self {
-        Self { data }
+    pub fn new(data: Option<(ContextId, PublicKey)>) -> Self {
+        Self {
+            data: data.map(|(context_id, member_public_key)| JoinContextResponseData {
+                context_id,
+                member_public_key,
+            }),
+        }
     }
 }
 
 #[derive(Clone, Copy, Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
-#[non_exhaustive]
 pub struct UpdateContextApplicationRequest {
     pub application_id: ApplicationId,
 }
 
 impl UpdateContextApplicationRequest {
-    #[must_use]
     pub const fn new(application_id: ApplicationId) -> Self {
         Self { application_id }
     }
 }
 
 #[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
 pub struct UpdateContextApplicationResponse {
     pub data: Empty,
 }
 
+impl UpdateContextApplicationResponse {
+    pub const fn new() -> Self {
+        Self { data: Empty {} }
+    }
+}
+
+// -------------------------------------------- Identity API ----------------------------------------
 #[derive(Clone, Copy, Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
-#[non_exhaustive]
-pub struct GetContextStorageResponseData {
-    pub size_in_bytes: u64,
+pub struct GenerateContextIdentityResponseData {
+    pub public_key: PublicKey,
+    pub private_key: PrivateKey,
```

Comment:
we really shouldn't expose private keys at all through the server, I thought we could pre-save unassociated private keys to some record, say `ContextIdentity { context_id: [0;32] }` so when the invitation payload comes back, we'll have the private key, but that's not implemented yet